Make the expected method calling order in SurfaceTool more explicit

Calinou opened the pull request:
I bumped into this today while working on https://github.com/godotengine/godot/pull/40106.

Calinou committed:
Make the expected method calling order in SurfaceTool more explicit

clayjohn commented on doc/classes/SurfaceTool.xml:
```diff
@@ -47,7 +48,8 @@
 			<description>
-				Adds an index to index array if you are using indexed vertices. Does not need to be called before adding vertices.
+				Adds an index to index array if you are using indexed vertices.
+				[b]Note:[/b] If you call [method add_index], it must be called [i]before[/i] calling [method add_vertex]. Otherwise, it won't behave as desired.
```
The index array can be created after finishing with all vertices or while vertices are being submitted.

clayjohn commented on doc/classes/SurfaceTool.xml:
```diff
@@ -29,6 +29,7 @@
 			</argument>
 			<description>
 				Adds an array of bones for the next vertex to use. [code]bones[/code] must contain 4 integers.
+				[b]Note:[/b] This must be called [i]before[/i] calling [method add_vertex]. Otherwise, it won't behave as desired.
```
For these descriptions, I prefer to explain in more detail that this will be applied to the next vertex, if it is called after adding a vertex, it will be applied to the next vertex. If every vertex needs this and you fail to submit it for the first vertex, it may not be used at all. 

I don't like just saying it may not behave as desired.

clayjohn commented on doc/classes/SurfaceTool.xml:
```diff
@@ -66,6 +69,7 @@
 			</argument>
 			<description>
 				Specifies whether the current vertex (if using only vertex arrays) or current index (if also using index arrays) should use smooth normals for normal calculation.
+				[b]Note:[/b] This must be called [i]before[/i] calling [method add_vertex]. Otherwise, it won't behave as desired.
```
I think this can be called when creating the index array. Which means it can be called after all calls to add_vertex

clayjohn commented on doc/classes/SurfaceTool.xml:
```diff
@@ -104,6 +109,7 @@
 			</argument>
 			<description>
 				Specifies a set of UV coordinates to use for the next vertex.
+				[b]Note:[/b] This must be called [i]before[/i] calling [method add_vertex]. Otherwise, it won't behave as desired.
```
Don't think this applies here

Calinou commented:
I rebased this PR against the latest `master` branch and noticed that most methods I modified in the SurfaceTool class are no longer in the class reference. Why is that?

akien-mga commented:
@reduz renamed `add_` methods to `set_`.

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.3.
